crates/environ/src/trap_encoding.rs

```diff
@@ -87,6 +87,9 @@ pub enum Trap {
 
     /// Used to indicate that a trap was raised by atomic wait operations on non shared memory.
     AtomicWaitNonSharedMemory,
+
+    /// A debug assertion failed.
+    InternalAssertionFailed,
```

Comment:
Should this perhaps be `#[doc(hidden)]`? (this'll end up in the `wasmtime` crate)

Reply:
Or should we perhaps `abort()` if this ever gets seen?